code/modules/mapping/space_management/space_reservation.dm

```diff
@@ -0,0 +1,70 @@
+
+//Yes, they can only be rectangular.
+//Yes, I'm sorry.
+/datum/turf_reservation
+	var/list/reserved_turfs = list()
+	var/bottom_left_coords[3]
+	var/top_right_coords[3]
+	var/wipe_reservation_on_release = TRUE
+	var/turf_type = /turf/open/space
+
+/datum/turf_reservation/transit
+	turf_type = /turf/open/space/transit

```

Comment:
You already override turf_type when you call the proc

Reply:
@Cyberboss it's there because there isn't a real performance hit to including it and if something requests it without an override it'll default to transit if the something asked for a transit reservation, i can remove it if you want ofcourse.